src/classes/CRLP_Rollup_SEL_TEST.cls

```diff
@@ -48,59 +50,227 @@ public class CRLP_Rollup_SEL_TEST {
         String filterGroupsJSON = '[' + CMT_UnitTestData_TEST.createFilterGroupRecord(filterGroupId, 'TestFilterGroup') + ']';
         CMT_FilterRule_SEL.cachedFilterGroups = (List<Filter_Group__mdt>)JSON.deserialize(filterGroupsJSON, List<Filter_Group__mdt>.class);
 
-        // Create a single Rollup that uses the above Filter Group
+        // Create a series of Rollups that each specify a different Opp field in for the Rollup__mdt Detail and Amount
+        // fields that should never be included when building the Opportunity Query. This is defined in the Map
+        // in CRLP_Rollup_SEL.UNSUPPORTED_OPERATIONS_BY_ROLLUP_FIELD.
         String rollupsJSON = '[' +
-                CMT_UnitTestData_TEST.buildCMTObjectJSON('Rollup__mdt', 'Household Total Donations All Time',
-                    CMT_UnitTestData_TEST.getNewRecordId(), new List<CMT_UnitTestData_TEST.CMT_Field>{
-                        new CMT_UnitTestData_TEST.CMT_Field('Description__c', CMT_UnitTestData_TEST.CMTFieldType.FldText, 'Household Total Donations All Time'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Filter_Group__c', CMT_UnitTestData_TEST.CMTFieldType.FldText, filterGroupId),
-                        new CMT_UnitTestData_TEST.CMT_Field('Active__c', CMT_UnitTestData_TEST.CMTFieldType.FldBoolean, True),
-                        new CMT_UnitTestData_TEST.CMT_Field('Summary_Object__c', CMT_UnitTestData_TEST.CMTFieldType.FldEntity, 'Account'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Summary_Field__c', CMT_UnitTestData_TEST.CMTFieldType.FldField, 'Account.AnnualRevenue'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Detail_Object__c', CMT_UnitTestData_TEST.CMTFieldType.FldEntity, 'Opportunity'),
-                        /*new CMT_UnitTestData_TEST.CMT_Field('Detail_Field__c', CMT_UnitTestData_TEST.CMTFieldType.FldField, 'Opportunity.AccountId'),*/
-                        new CMT_UnitTestData_TEST.CMT_Field('Date_Object__c', CMT_UnitTestData_TEST.CMTFieldType.FldEntity, 'Opportunity'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Date_Field__c', CMT_UnitTestData_TEST.CMTFieldType.FldField, 'Opportunity.CloseDate'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Amount_Object__c', CMT_UnitTestData_TEST.CMTFieldType.FldEntity, 'Opportunity'),
-                        new CMT_UnitTestData_TEST.CMT_Field('Amount_Field__c', CMT_UnitTestData_TEST.CMTFieldType.FldField, 'Opportunity.Amount')
-                }) + ']';
+
+            // Sum operation. Ignores the Detail field reference
+            CMT_UnitTestData_TEST.createRollupRecord('Household Total Donations', filterGroupId,
```

Comment:
It would be nice to have CRLP testing API. I think this test data setup has been used before so we might have this common code extracted into test API in another WI.

Reply:
Yeah, we had talked about that once before. Basically creating a nicely abstracted class to mock CMT records. I started looking at it earlier today while starting refactors related to the API, but there are a **lot** of references to the `createRollupRecord` method that would require refactoring. Not sure it's worth tackling at this point.